### ISSUE TITLE
support for records

### ISSUE THREAD

**Issue by vmachacek** [mostmand/Cloneable#13]
please add support for records, 

now this wont do anything

```
[Cloneable.Cloneable]
public partial record Foo(string Bar);
```

**Comment by Spoomer** [comment]
I don't know, if you know this, but you can copy records with the `with` keyword:

`Foo with {}`

**Comment by vmachacek** [comment]
yes, i figured this after i posted this, but that will only do shallow copy, correct?

**Comment by Spoomer** [comment]
Correct! Records should be optimally immutable - so this should be ok, if you don't use mutable classes as properties..
